src/metrics.go

```diff
@@ -71,25 +73,11 @@ func (mc *metricsCollector) collect() {
 
 	metricChan := make(chan newrelicMetricSender, bufferSize) // large buffer for speed
 
-	// Create a goroutine for each of the tablespace metric groups to collect
-	if tablespaceWhiteList != nil && len(tablespaceWhiteList) == 0 {
-		log.Info("No tablespaces specified, skipping tablespace collection.")
-	} else {
-		for _, tbCollection := range tablespaceCollections {
-			if mc.skipGroup(tbCollection.name) {
-				log.Debug("Metric group %s skipped.", tbCollection.name)
-				continue
-			}
-			collectorWg.Add(1)
-			c := tbCollection
-			go c.Collect(mc.db, &collectorWg, metricChan)
-		}
-	}
-
-	// Collect tablespace first so the list query completes before other queries are run
-	collectorWg.Wait()
+	// Separate logic is needed to see if we should even collect tablespaces
+	// Collect tablespaces first so the list query completes before other queries are run
+	collectorWg.Add(1)
+	go collectTableSpaces(mc.db, &collectorWg, metricChan, tablespaceCollections)
```

Comment:
Ass discussed, we need it because `collect` should nor block here, thanks a lot  for checking @carlossscastro

Reply:
It's true, this was like this previously but it didn't make any sense to have it in a go routine since the waitGroup will already wait for the ones spawned.